packages/components/src/components/modal/modal.jsx

```diff
@@ -39,11 +39,11 @@ class ModalElement extends React.PureComponent {
 
     handleClickOutside = event => {
         const path = event.path || (event.composedPath && event.composedPath());
-
         if (
             this.props.has_close_icon &&
             this.wrapper_ref &&
             !path.some(el => el === this.wrapper_ref) &&
+            !(this.props.elements_to_ignore && path.find(el => this.props.elements_to_ignore.includes(el))) &&
```

Comment:
I don't understand what this has to do with scrolling 🤔 Could you explain it @nijil-binary ? A description to the PR would be helpful.

Reply:
Sorry about that. When the modal is open and you click on the preview workspace, `event.target` for some reason is not the preview workspace but the main bot workspace which is not part of `wrapper_ref`. Hence the modal gets dismissed every time I click on it. So, I had to pass the main workspace as a prop separately in a more generalized way and put an explicit check for it here.

Reply:
Thanks I see, so the event.target is what's behind the modal and therefore onClickOutside is fired and the modal closes?

Reply:
Yes, that's right.

Reply:
strange, any idea why the modal is being clicked through? @zaki-binary @negar-binary

Reply:
@oskar-binary that's because the workspace is injected after modal mounted. So, the workspace won't be part of the `wrapper_ref`.